src/main/main.ts

```diff
@@ -188,214 +189,195 @@ function initializeApp() {
   });
 }
 
-function initializeIpc() {
-  ipcMain.on(
-    "encounter different version",
-    async (event, data: DifferentAppProtocolVersionEncounterSubscription) => {
-      const localVersionNumber =
-        data.differentAppProtocolVersionEncounter.localVersion.version;
-      const peerVersionNumber =
-        data.differentAppProtocolVersionEncounter.peerVersion.version;
-      if (peerVersionNumber <= localVersionNumber) {
-        console.log(
-          "Encountered version is not higher than the local version. Abort update."
-        );
-        return;
-      }
+async function update(
+  localVersionNumber: number,
+  peerVersionNumber: number,
+  peerVersionExtra: string
+) {
+  if (peerVersionNumber <= localVersionNumber) {
+    console.log(
+      "Encountered version is not higher than the local version. Abort update."
+    );
+    return;
+  }
 
-      if (lockfile.checkSync(lockfilePath)) {
-        console.log(
-          "'encounter different version' event seems running already. Stop this flow."
-        );
-        return;
-      } else {
-        try {
-          lockfile.lockSync(lockfilePath);
-          console.log(
-            "Created 'encounter different version' lockfile at ",
-            lockfilePath
-          );
-        } catch (e) {
-          console.error("Error occurred during trying lock.");
-          throw e;
-        }
-      }
+  if (lockfile.checkSync(lockfilePath)) {
+    console.log(
+      "'encounter different version' event seems running already. Stop this flow."
+    );
+    return;
+  } else {
+    try {
+      lockfile.lockSync(lockfilePath);
+      console.log(
+        "Created 'encounter different version' lockfile at ",
+        lockfilePath
+      );
+    } catch (e) {
+      console.error("Error occurred during trying lock.");
+      throw e;
+    }
+  }
```

Comment:
There is no need to use `else`.
```suggestion

  try {
    lockfile.lockSync(lockfilePath);
    console.log(
      "Created 'encounter different version' lockfile at ",
      lockfilePath
    );
  } catch (e) {
    console.error("Error occurred during trying lock.");
    throw e;
  }
```